src/ynEIGEN/RedemptionAssetsVault.sol

```diff
@@ -0,0 +1,227 @@
+
+    IynEigen public ynEigen;
+    IAssetRegistry public assetRegistry;
+    bool public paused;
+    address public redeemer;
+
+    mapping(address asset => uint256 balance) public balances;
+
+    // Initializer with Init struct and roles
+    struct Init {
+        address admin;
+        address redeemer;
+        IynEigen ynEigen;
+        IAssetRegistry assetRegistry;
+    }
+
+    function initialize(Init memory init)
+        external
+        notZeroAddress(init.admin)
+        notZeroAddress(init.redeemer)
+        notZeroAddress(address(init.ynEigen))
+        notZeroAddress(address(init.assetRegistry))
+        initializer {
+        __AccessControl_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, init.admin);
+        _grantRole(PAUSER_ROLE, init.admin);
+        _grantRole(UNPAUSER_ROLE, init.admin);
+
+        redeemer = init.redeemer;
+        ynEigen = init.ynEigen;
+        assetRegistry = init.assetRegistry;
+        paused = false;
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  REDEMPTION  --------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    function deposit(uint256 amount, address asset) external {
+        if (!assetRegistry.assetIsSupported(IERC20(asset))) revert AssetNotSupported();
+
+        balances[asset] += amount;
+        IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
+
+        emit AssetDeposited(asset, msg.sender, amount);
+    }
+
+    /** 
+     * @notice Calculates the current redemption rate of ynETH to ETH.
+     * @return The current redemption rate as a uint256.
+     */
+    function redemptionRate() public view returns (uint256) {
+        return ynEigen.previewRedeem(YNETH_UNIT);
+    }
+
+    /** 
+     * @notice Returns the total amount of assets available for redemption.
+     * @return _availableRedemptionAssets The available unit-of-account-denominated balance as a uint256.
+     */
+    function availableRedemptionAssets() public view returns (uint256 _availableRedemptionAssets) {
+
+        IERC20[] memory assets = assetRegistry.getAssets();
+
+        uint256 len = assets.length;
+        for (uint256 i = 0; i < len; ++i) {
+            IERC20 asset = assets[i];
+            uint256 balance = balances[address(asset)];
+            if (balance > 0) _availableRedemptionAssets += assetRegistry.convertToUnitOfAccount(asset, balance);
+        }
+    }
+
+    /** 
+     * @notice Transfers a specified amount of redemption assets to a given address.
+     * @param to The recipient address of the assets.
+     * @param amount The amount of assets to transfer.
+     * @dev Requires the caller to be the redeemer and the contract to not be paused.
+     */
+    function transferRedemptionAssets(address to, uint256 amount, bytes calldata /* data */) public onlyRedeemer whenNotPaused nonReentrant {
```

Comment:
WE can move forward with strategy and explain the  distribution strategy in the natspec

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/164/commits/6e73ce734c4fa2cabaf231331c38b642e6358f6f